### PR TITLE
Take advantage of the Vert.x Context API improvements

### PR DESCRIPTION
**Pull request by Sanne**
Follows up on https://github.com/eclipse-vertx/vert.x/pull/4073

### CODE REVIEW

**Comment by gavinking** [comment]
ah yes, i had forgotten about this

**Review by gavinking — APPROVED** [review]

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-reactive/pull/948

Thanks

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/context/impl/VertxContext.java [hunk 1]
```diff
@@ -50,9 +46,9 @@ static ContextInternal currentContext() {
     @Override
     @SuppressWarnings("unchecked")
     public <T> T get(Key<T> key) {
-        final ContextInternal context = currentContext();
+        final io.vertx.core.Context context = Vertx.currentContext();
         if ( context != null ) {
-            return (T) context.localContextData().get( key );
+            return (T) context.get( key );
```
Comment (gavinking) [inline hibernate-reactive-core/src/main/java/org/hibernate/reactive/context/impl/VertxContext.java:51]: @Sanne why did you change this to use the main context instead of the local context?

Isn't that wrong for what we're using it for?

Comment (Sanne) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/context/impl/VertxContext.java:51]: yes it's wrong - it wasn't intentional, I wasn't aware of the "global context" and all tests passed fine.

Ironically my benchmarks also showed no issue, as we don't use the APIs which use the context. Sorry :(

Comment (gavinking) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/context/impl/VertxContext.java:51]: @Sanne, we all share in the blame for this one. I approved it, Davide merged it. I *certainly* should have noticed but I just didn't read the diff carefully enough.

It's embarrassing but that's a think that can happen with this stuff that's hard to test. Hopefully we can figure out some way to test it.
